### PR TITLE
create GetEmployeeInfo and change HoursWorked output format

### LINKED ISSUE #11: GET: `/api/manger/employees?id=`
## API description
- URL queries: (optional) id
- Authen header: Required.
- Body: None
- Description: This API will return this manager's employees' information. If `id` query is passed in, first check if the manager manages that employee, if not, return error 403, otherwise return the information of that employee. Django View permission_class may be used here. Please refer to Django docs for more detail.

### PR DESCRIPTION
**Pull request by mimiramita**
I created GetEmployeeInfo and changed the HoursWorked output format by adding "status":"success. #11

### CODE REVIEW

File: account/views.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ def get(self, request):
                 for time_entry in time_entries:
                     results.append(
                         {"Date": time_entry.date, "Hours": time_entry.hoursWorked})
-                return Response(results, status=status.HTTP_200_OK)
+                return Response({'status': 'success', 'Response': results}, status=status.HTTP_200_OK)
```
Comment (nguyenvothuan) [inline account/views.py:105]: nice!

File: account/views.py [hunk 2]
```diff
@@ -146,6 +146,48 @@ def get(self, request):
             return Response({'status': 'success', 'Response': {'email':email, 'first_name':first_name, 'last_name':last_name, 'company':company, 'manager':manager}}, 
                                 status=status.HTTP_200_OK)
 
+        except Exception as e:
+            return Response({'status': False, 'message': str(e)},
+                            status=status.HTTP_400_BAD_REQUEST)
+
+employee_id = openapi.Parameter('id', in_=openapi.IN_QUERY,
```
Comment (nguyenvothuan) [inline account/views.py]: try to include `employee_id` inside the `GetEmployeeInfo`'s APIView if you don't plan to reuse `employee_id` anywhere else. It's just a way to not pollute the global scope with these variables.

File: account/views.py [hunk 3]
```diff
@@ -146,6 +146,48 @@ def get(self, request):
...
+                         type=openapi.TYPE_STRING)   
+
+class GetEmployeeInfo(APIView):
+    @swagger_auto_schema(
+        manual_parameters=[employee_id],
```
Comment (nguyenvothuan) [inline account/views.py:168]: you can simply define `employee_id` inside this schema.

File: account/views.py [hunk 4]
```diff
@@ -146,6 +146,48 @@ def get(self, request):
...
+
+            employee_id = request.GET.get('id')
+            if employee_id != None:
+                employee_accounts = Account.objects.filter(id=employee_id)
+                if len(employee_accounts) == 0:
+                    return Response({'status': False, 'message': "Invalid employee id"},
+                                    status=status.HTTP_403_FORBIDDEN)
+                employee_account = employee_accounts[0]
+
+                if employee_account.manager == None or employee_account.manager.id != manager_id:
```
Comment (nguyenvothuan) [inline account/views.py:184]: yh, this check makes sense

File: account/views.py [hunk 5]
```diff
@@ -146,6 +146,48 @@ def get(self, request):
...
+                    return Response({'status': False, 'message': "Data is forbidden"},
+                                    status=status.HTTP_403_FORBIDDEN)
+
+                return Response({'status': 'success', 'Response': {'email': employee_account.user.email, 'first_name': employee_account.user.first_name, 'last_name': employee_account.user.last_name, 'company': employee_account.company.name}},
```
Comment (nguyenvothuan) [inline account/views.py]: As I said earlier in the prev class, you can return the whole object's data with a single serializer. Check it out. You can still compose a dictionary and return it like this, but the problem is it doesn't look that concise though. It's up to you to decide. I know finishing the task asap is the main priority so I don't want to stress this too much.

File: account/views.py [hunk 6]
```diff
@@ -146,6 +146,48 @@ def get(self, request):
...
+                employee_account = employee_accounts[0]
+
+                if employee_account.manager == None or employee_account.manager.id != manager_id:
+                    return Response({'status': False, 'message': "Data is forbidden"},
+                                    status=status.HTTP_403_FORBIDDEN)
+
+                return Response({'status': 'success', 'Response': {'email': employee_account.user.email, 'first_name': employee_account.user.first_name, 'last_name': employee_account.user.last_name, 'company': employee_account.company.name}}, 
+                                status=status.HTTP_200_OK)
+                
+            employees = Account.objects.filter(manager=account)[:20]
```
Comment (nguyenvothuan) [inline account/views.py:192]: so this is to handle the case when there is no `employee_id` passed in? Cool!

File: account/views.py [hunk 7]
```diff
@@ -146,6 +146,48 @@ def get(self, request):
...
+            if employees == None:
+                    return Response({'status': False, 'message': "No employees for this user"},
+                                    status=status.HTTP_403_FORBIDDEN)
+            results = []
+            for employee in employees:
+                    results.append(
+                        {'email': employee.user.email, 'first_name': employee.user.first_name, 'last_name': employee.user.last_name, 'company': employee.company.name})
```
Comment (nguyenvothuan) [inline account/views.py]: Again, i would like to recommend using a serializer for this. Isn't it a way to learn more about Django :D?

**Review by nguyenvothuan — CHANGES_REQUESTED** [review]
Nice job! Just some feedback from my end. Also, after you have fixed all the stuff mentioned here, it's really nice to have a recording to show the demo of your API working properly.

**Comment by mimiramita** [comment]
https://user-images.githubusercontent.com/111351821/235571580-7429dad3-bf49-41f7-879b-3dd4ac14ab6c.mov

**Comment by nguyenvothuan** [comment]
LGTM

**Review by nguyenvothuan — APPROVED** [review]

**Pull request merged**